src/page-weekly-snapshot/projectionChartUtils.tsx

```diff
@@ -0,0 +1,187 @@
+import * as dateFns from "date-fns";
+import ndarray from "ndarray";
+
+import {
+  EpidemicModelInputs,
+  getLocaleDefaults,
+  RateOfSpread,
+  totalConfirmedCases,
+  totalConfirmedDeaths,
+} from "../impact-dashboard/EpidemicModelContext";
+import { countCasesForDay } from "../impact-dashboard/ImpactProjectionTableContainer";
+import { NUM_DAYS } from "../infection-model";
+import { calculateCurves, curveInputsFromUserInputs } from "../infection-model";
+import { getAllValues, getColView } from "../infection-model/matrixUtils";
+import { seirIndex } from "../infection-model/seir";
+import { LocaleData } from "../locale-data-context";
+import { ModelInputs } from "../page-multi-facility/types";
+import { combineFacilitiesProjectionData } from "../page-response-impact/responseChartData";
+
+export function getCaseCount(curveData: ndarray<number>) {
+  const cases = [];
+  for (let day = 0; day < curveData.shape[0]; ++day) {
+    const caseCountForDay = countCasesForDay(curveData, day);
+    cases.push(caseCountForDay);
+  }
+  return cases;
+}
+
+export const today = dateFns.endOfToday();
+export const ninetyOneDaysAgo = dateFns.subDays(today, 91);
+export const ninetyDaysAgo = dateFns.subDays(today, 90);
+
+function addProjectionPadding(values: number[], numDays = 90): number[] {
+  return [...Array(numDays - values.length).fill(0), ...values];
+}
+
+export function getVersionForProjection(versions: ModelInputs[]) {
+  const versionsWithCases = versions
+    .sort(({ observedAt: a }, { observedAt: b }) => dateFns.compareAsc(a, b))
+    .filter((version) => totalConfirmedCases(version) > 0);
+
+  const version90DaysAgo = versionsWithCases.find((v) =>
+    dateFns.isSameDay(v.observedAt, ninetyDaysAgo),
+  );
+
+  let versionForProjection: ModelInputs | undefined = version90DaysAgo;
+
+  // If the version from 90 days ago has 0 or no data, look forward for the next
+  // version with cases
+  if (!version90DaysAgo) {
+    const closestVersionWithCases = versionsWithCases.find((version) => {
+      const date = dateFns.closestTo(
+        ninetyDaysAgo,
+        versionsWithCases.map((v) => v.observedAt),
+      );
+      return dateFns.isSameDay(version.observedAt, date);
+    });
+    versionForProjection = closestVersionWithCases;
+  }
+  return versionForProjection;
+}
+
+export function getProjectedData(epidemicModelInputs: EpidemicModelInputs) {
+  if (!epidemicModelInputs || !epidemicModelInputs?.observedAt) {
```

Comment:
There are no expected cases where this conditional would be true and _not_ be exceptional? e.g. for a brand new baseline scenario?

Reply:
Yeah I think it is possible for a new baseline scenario to be shared without any cases/version history, I wasn't sure yet at this point what to do in that case, so I updated this to return zero for each day of the projected line. If there's another strategy that would make more sense in this case let me know, happy to talk it through too.